### DISCUSSION TITLE
Docker windows bash error

### DISCUSSION THREAD

**Discussion by DrHybris** [mdolab/dafoam#479] [Bug]
I did the follwing:
1) Go to workshops git clone
`cd .\DAFoam\workshops\2021_Summer\examples\naca0012\incompressible\`
2) Run docker images using 
`docker run -it --rm -u dafoamuser --mount "type=bind,src=$(pwd),target=/home/dafoamuser/mount" -w /home/dafoamuser/mount dafoam/opt-packages:v3.0.7 bash`
pd: Im in windows and i replace %cd% with $(pwd) because this work for me for run the docker image intead of %cd%
3) following the tutorial i begin with
`./preProcessing.sh`

and i get the next error:
`➜ docker run -it --rm -u dafoamuser --mount "type=bind,src=$(pwd),target=/home/dafoamuser/mount" -w /home/dafoamuser/mount dafoam/opt-packages:v3.0.7 bash
dafoamuser@fb3ee39343e0:~/mount$ ls
0.orig  Allclean.sh  FFD  constant  genAirFoilMesh.py  paraview.foam  preProcessing.sh  profiles  runScript.py  system
dafoamuser@fb3ee39343e0:~/mount$ ./preProcessing.sh
bash: ./preProcessing.sh: /bin/bash^M: bad interpreter: No such file or directory
dafoamuser@fb3ee39343e0:~/mount$
`
Im using windows terminal(Windows powershell)

**Comment by friedenhe** [comment]
Can you copy the tutorial inside the docker container and run it? For example copy it to `/home/dafoamuser/` The error you saw is likely related to the file format incompatibility between Windows and Linux. Also, you can't use DAFoam 3.0.7 to run the workshop cases.  You need to use an old version, check the first workshop slide.

**Reply by DrHybris** [reply]
The Docker work well with the example in the workshop 2022, when i have tu run the python scrip `runScrip`, the problem is when i have ti use ./ for execute .sh

**Reply by friedenhe** [reply]
run `chmod -R 750 *`
